src/main/java/com/yahoo/bullet/operations/aggregations/sketches/QuantileSketch.java

```diff
@@ -0,0 +1,287 @@
+     * points (evenly-spaced).
+     *
+     * @param k A number representative of the size of the sketch.
+     * @param type A {@link DistributionType} that determines what the points mean.
+     * @param numberOfPoints A positive number of evenly spaced points in the range for the type to get the data for.
+     */
+    public QuantileSketch(int k, DistributionType type, int numberOfPoints) {
+        this(k, type);
+        this.numberOfPoints = numberOfPoints;
+    }
+
+    /**
+     * Updates the sketch with a double.
+     *
+     * @param data A double to insert into the sketch.
+     */
+    public void update(double data) {
+        updateSketch.update(data);
+        updated = true;
+    }
+
+    @Override
+    public void union(byte[] serialized) {
+        DoublesSketch sketch = DoublesSketch.heapify(new NativeMemory(serialized));
+        unionSketch.update(sketch);
+        unioned = true;
+    }
+
+    @Override
+    public byte[] serialize() {
+        collect();
+        return merged.toByteArray();
+    }
+
+    @Override
+    public Clip getResult(String metaKey, Map<String, String> conceptKeys) {
+        Clip data = super.getResult(metaKey, conceptKeys);
+        double[] domain = getDomain();
+        double[] range;
+        if (type == DistributionType.QUANTILE) {
+            range = merged.getQuantiles(domain);
+        } else if (type == DistributionType.PMF) {
+            range = merged.getPMF(domain);
+        } else {
+            range = merged.getCDF(domain);
+        }
+        data.add(zip(domain, range, type, getNumberOfEntries()));
+        return data;
+    }
+
+    @Override
+    protected void collect() {
+        if (updated && unioned) {
+            unionSketch.update(updateSketch);
+        }
+        merged = unioned ? unionSketch.getResult() : updateSketch.compact();
+    }
+
+    @Override
+    public void reset() {
+        unioned = false;
+        updated = false;
+        updateSketch.reset();
+        unionSketch.reset();
+    }
+
+    @Override
+    protected Map<String, Object> getMetadata(Map<String, String> conceptKeys) {
+        Map<String, Object> metadata = super.getMetadata(conceptKeys);
+
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.MINIMUM_VALUE.getName()), this::getMinimum);
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.MAXIMUM_VALUE.getName()), this::getMaximum);
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.ITEMS_SEEN.getName()), this::getNumberOfEntries);
+        addIfKeyNonNull(metadata, conceptKeys.get(Concept.NORMALIZED_RANK_ERROR.getName()), this::getNormalizedRankError);
+
+        return metadata;
+    }
+
+    @Override
+    protected Boolean isEstimationMode() {
+        return merged.isEstimationMode();
+    }
+
+    @Override
+    protected String getFamily() {
+        return Family.QUANTILES.getFamilyName();
+    }
+
+    @Override
+    protected Integer getSize() {
```

Comment:
No, they are protected or private not public and are meant to be called internally or by subclasses only.